drivers/gpio/Kconfig.nrfx

```diff
@@ -31,5 +31,29 @@ config GPIO_NRF_P1
 	help
 	  Enable nRF GPIO port P1 config options.
 
+choice
+	prompt "nRF GPIO edge interrupts mechanism"
+	default GPIO_NRF_INT_EDGE_USING_GPIOTE
+
+config GPIO_NRF_INT_EDGE_USING_GPIOTE
```

Comment:
Any reason to have this as a Kconfig options versus just always supporting it?

Reply:
When I considered adding this I planned to have it be selected on a per-instance basis by a driver-specific configuration flag, but Zephyr isn't really prepared for that level of vendor-specific control.

Reply:
Selecting one or the other option will be based on 1) application, 2) platform (only some nRF platforms are affected by power hungry GPIOTE). It also doesn't make sense to have one mechanism (e.g. gpio sense) for one GPIO instance, second mechanism (gpiote) for second GPIO instance. Implementation would also bloat too much, both in source as well in output size. So an alternative to Kconfig is device-tree. We could use that, but to keep things simple we should enforce the same configuration for each (enabled) instance.